tests/Add-DbaAgDatabase.Tests.ps1

```diff
@@ -38,49 +47,130 @@ Describe "Add-DbaAgDatabase" -Tag "UnitTests" {
     }
 }
 
-Describe "Add-DbaAgDatabase" -Tag "IntegrationTests" {
+Describe "Add-DbaAgDatabase IntegrationTests" -Tag "IntegrationTests" {
     BeforeAll {
-        $null = Get-DbaProcess -SqlInstance $TestConfig.instance3 -Program 'dbatools PowerShell module - dbatools.io' | Stop-DbaProcess -WarningAction SilentlyContinue
-        $server = Connect-DbaInstance -SqlInstance $TestConfig.instance3
-        $agname = "dbatoolsci_addagdb_agroup"
-        $dbname = "dbatoolsci_addagdb_agroupdb"
-        $newdbname = "dbatoolsci_addag_agroupdb_2"
-        $server.Query("create database $dbname")
-        $backup = Get-DbaDatabase -SqlInstance $TestConfig.instance3 -Database $dbname | Backup-DbaDatabase
+        # We want to run all commands in the BeforeAll block with EnableException to ensure that the test fails if the setup fails.
+        $PSDefaultParameterValues['*-Dba*:EnableException'] = $true
+
+        # We remove all the calls to Get-DbaProcess and Stop-DbaProcess as they are not needed with correct tests.
+        #$null = Get-DbaProcess -SqlInstance $TestConfig.instance3 -Program 'dbatools PowerShell module - dbatools.io' | Stop-DbaProcess -WarningAction SilentlyContinue
+
+        # Collect all the created files to be able to remove them ant the end.
+        $filesToRemove = @( )
+
+        # Connect to the instance(s) using Connect-DbaInstance only if needed.
+        # It is saver to use the instance name from $TestConfig to get a fresh new SMO every time.
+        # In the future, we should also test with pre-opened connection.
+        # Use dbatools commands whenever possible, so that these are tested here as well.
+
+        # Explain what needs to be set up for the test:
+        # To add a database to an availablity group, we need an availability group and a database that has been backed up.
+        # For negative tests, we need a database without a backup and a non existing database.
+
+        # Set variables that we need in the tests.
+        # I don't see the point in prefixing them all with "dbatoolsci",
```

Comment:
someone may want to test specific functions with a "not superclean" environment. A prefix makes the artifacts generated readily identifyiable also at a later stage

Reply:
I see the cons of that - but only in those "not superclean" environments. And I thought about that. And I came to the conclusion that it is super dangerous to run our tests in any other than a dedicated environment. And in case of a new environment we don't need individual but nice names.
But if I'm the only one on that side - I can totally live with "dbatoolsci_" and a lot of "..._$(Get-Random)".